[CI] DenseVectorFieldMapperTests testKnnQuantizedFlatVectorsFormat failing

**Build Scans:**
- [elasticsearch-periodic #11473 / release-tests](https://gradle-enterprise.elastic.co/s/2g3grhe3wsb6m)
- [elasticsearch-pull-request #105215 / part-1](https://gradle-enterprise.elastic.co/s/7q2wcmb4rcyf4)
- [elasticsearch-pull-request #105212 / checkpart1_release-tests](https://gradle-enterprise.elastic.co/s/rfcf4zcezkq4c)
- [elasticsearch-pull-request #105179 / part-1](https://gradle-enterprise.elastic.co/s/7lyg5ewltxu5i)
- [elasticsearch-periodic-java-ea #89 / release-tests](https://gradle-enterprise.elastic.co/s/levtsi7sxpuow)
- [elasticsearch-pull-request #105045 / checkpart1_release-tests](https://gradle-enterprise.elastic.co/s/45pvha7muaiak)
- [elasticsearch-pull-request #105023 / checkpart1_release-tests](https://gradle-enterprise.elastic.co/s/b6gjkaycsl6tw)
- [elasticsearch-pull-request #104990 / checkpart1_release-tests](https://gradle-enterprise.elastic.co/s/yznqkesmlikqm)
- [elasticsearch-pull-request #104990 / checkpart1_release-tests](https://gradle-enterprise.elastic.co/s/trzzgam7i5dwu)
- [elasticsearch-pull-request #104532 / part-1](https://gradle-enterprise.elastic.co/s/a5qyc24n7l4cs)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.index.mapper.vectors.DenseVectorFieldMapperTests.testKnnQuantizedFlatVectorsFormat" -Dtests.seed=1841EE87ADFCD772 -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=fr-WF -Dtests.timezone=SystemV/AST4ADT -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.index.mapper.vectors.DenseVectorFieldMapperTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testKnnQuantizedFlatVectorsFormat'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: with toString() "ES813Int8FlatVectorFormat(name=ES813Int8FlatVectorFormat, innerFormat=ES814ScalarQuantizedVectorsFormat(name=ES814ScalarQuantizedVectorsFormat, confidenceInterval=null, bits=7, compressed=false, flatVectorScorer=ESQuantizedFlatVectorsScorer(delegate=ScalarQuantizedVectorScorer(nonQuantizedDelegate=DefaultFlatVectorScorer()), factory=org.elasticsearch.simdvec.VectorScorerFactoryImpl@138348cd), rawVectorFormat=Lucene99FlatVectorsFormat(vectorsScorer=DefaultFlatVectorScorer())))"
     but: toString() was "ES813Int8FlatVectorFormat(name=ES813Int8FlatVectorFormat, innerFormat=ES814ScalarQuantizedVectorsFormat(name=ES814ScalarQuantizedVectorsFormat, confidenceInterval=null, bits=7, compressed=false, flatVectorScorer=ESFlatVectorsScorer(delegate=ScalarQuantizedVectorScorer(nonQuantizedDelegate=DefaultFlatVectorScorer()), factory=org.elasticsearch.simdvec.VectorScorerFactoryImpl@138348cd), rawVectorFormat=Lucene99FlatVectorsFormat(vectorsSc
[truncated]
```

**Issue Reasons:**
- [main] 2 consecutive failures in step release-tests
- [main] 4 consecutive failures in step checkpart1_release-tests
- [main] 11 failures in test testKnnQuantizedFlatVectorsFormat (2.2% fail rate in 496 executions)
- [main] 2 failures in step release-tests (40.0% fail rate in 5 executions)
- [main] 4 failures in step part-1 (1.2% fail rate in 324 executions)
- [main] 5 failures in step checkpart1_release-tests (35.7% fail rate in 14 executions)
- [main] 8 failures in pipeline elasticsearch-pull-request (2.5% fail rate in 323 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -411,9 +411,6 @@ tests:
 - class: org.elasticsearch.xpack.ml.integration.RegressionIT
   method: testTwoJobsWithSameRandomizeSeedUseSameTrainingSet
   issue: https://github.com/elastic/elasticsearch/issues/138319
-- class: org.elasticsearch.index.mapper.vectors.DenseVectorFieldMapperTests
-  method: testKnnQuantizedFlatVectorsFormat
-  issue: https://github.com/elastic/elasticsearch/issues/138368
 - class: org.elasticsearch.smoketest.SmokeTestMultiNodeClientYamlTestSuiteIT
   method: test {yaml=termvectors/30_realtime/Realtime Term Vectors}
   issue: https://github.com/elastic/elasticsearch/issues/138370

--- a/server/src/test/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapperTests.java
+++ b/server/src/test/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapperTests.java
@@ -2003,7 +2003,7 @@ public class DenseVectorFieldMapperTests extends SyntheticVectorsMapperTestCase 
                     + (quantizedFlatFormat.equals("int4_flat") ? 4 : 7)
                     + ", compressed="
                     + quantizedFlatFormat.equals("int4_flat")
-                    + ", flatVectorScorer=ESQuantizedFlatVectorsScorer("
+                    + ", flatVectorScorer=ESFlatVectorsScorer("
                     + "delegate=ScalarQuantizedVectorScorer(nonQuantizedDelegate=DefaultFlatVectorScorer())"
                     + ", factory="
                     + (factory != null ? factory : "null")

